paasta_tools/mesos_tools.py

```diff
@@ -111,29 +111,29 @@ class MesosSlaveConnectionError(Exception):
     pass
 
 
-def get_mesos_leader(hostname=MY_HOSTNAME):
-    """Get the current mesos-master leader's hostname. Raise
-    MesosMasterConnectionError if we can't connect.
+def get_mesos_leader():
+    """Get the current mesos-master leader's hostname.
 
-    :param hostname: The hostname to query mesos-master on
     :returns: The current mesos-master hostname"""
-    redirect_url = 'http://%s:%s/redirect' % (hostname, MESOS_MASTER_PORT)
-    try:
-        r = requests.get(redirect_url, timeout=10)
-    except requests.exceptions.ConnectionError as e:
-        # Repackage the exception so upstream code can handle this case without
-        # knowing our implementation details.
-        raise MesosMasterConnectionError(repr(e))
-    r.raise_for_status()
-    return re.search('(?<=http://)[0-9a-zA-Z\.\-]+', r.url).group(0)
+    url = master.CURRENT.host
+    print "URL: %s" % url
+    print "CFG: %s" % os.environ['MESOS_CLI_CONFIG']
+    m = re.search('^https?://(.*?):\d+$', url)
+    if m:
+        ip = m.group(1)
+        host = socket.gethostbyaddr(ip)[0]
+        fqdn = socket.getfqdn(host)
+        return fqdn
+    else:
+        raise MesosMasterConnectionError()
```

Comment:
this is the wrong Exception to throw - this is raised when a regex doesn't match, not when you can't connect to the master.